static/app/components/compactSelect/composite.tsx

```diff
@@ -200,6 +200,18 @@ const RegionsWrap = styled('div')`
   overflow: auto;
   padding: ${space(0.5)} 0;
 
+  /* Add 1px to top padding if preceded by menu header, to account for the header's
+  shadow border */
+  div[data-select-overlay][data-has-header='true'] > div > & {
+    padding-top: calc(${space(0.5)} + 1px);
+  }
+
+  /* Add 1px to bottom padding if succeeded by menu footer, to account for the footer's
+  shadow border */
+  div[data-select-overlay][data-has-footer='true'] > div > & {
```

Comment:
Any way to make this less brittle? Could we instead pass these down as props?

Reply:
Yea I decided to do this because the `List` component also needs to know whether a footer/header is present in the overlay. `List` isn't directly used here in `control.tsx` — it's used in `index.tsx` and `composite.tsx`. So, taking the props only approach, we'd have to manually pass the props (e.g. `menuHasFooter` & `menuHasHeader`) down to three different locations in total. 

In contrast, with this approach, we need to define the data attributes only once, on `StyledOverlay`, and all of its descendants will have access to them via CSS attribute selectors. Since the data is defined in only one location, I think there is less chance of things breaking in the future.

You do have a point about brittleness — part of what makes this look brittle might be the double child selector (`… > div > &`). I wrote it like that because `CompactSelect` and `CompositeSelect` have slightly different DOM structures, so I needed the specificity to ensure that the CSS block applies only to the former and not the latter. Lemme see if there's a better way to structure the two components and simplify the selectors a bit.